### PR TITLE
Add scalars link

### PR DESCRIPTION
**Pull request by tgorgol**
**Description**

Changes proposed in this pull request:

- Add scalars link to Apollo Client that serializes and deserializes Labels type

**Related issue(s)**

kyma-incubator/compass#845

### CODE REVIEW

File: core-ui/src/index.js [hunk 1]
```diff
@@ -34,9 +34,9 @@ function setupLocalStorageVariables() {
 }
 
 preloadingStrategy(async () => {
-  builder.addEventListeners(() => {
+  builder.addEventListeners(async () => {
```
Comment (pkosiec) [inline core-ui/src/index.js]:
```suggestion
  await builder.addEventListeners(async () => {
```

File: core-ui/src/commons/builder.js [hunk 2]
```diff
@@ -7,19 +7,19 @@ class Builder {
   token = null;
   backendModules = [];
 
-  addEventListeners(callback) {
-    LuigiClient.addInitListener(e => {
+  async addEventListeners(callback) {
```
Comment (pkosiec) [inline core-ui/src/commons/builder.js]: hmm, I think we don't need `async` here after all

File: core-ui/src/apollo.js [hunk 3]
```diff
@@ -39,14 +41,33 @@ export function createCompassApolloClient() {
...
       dataIdFromObject: object => object.name || null,
     }),
   });
 }
 
+async function createScalarsLink(httpLink) {
+  const typesMap = {
+    Labels: {
+      serialize: parsed => JSON.stringify(parsed),
+      parseValue: raw => JSON.parse(raw),
```
Comment (Wawrzyn321) [inline core-ui/src/apollo.js]:
```suggestion
      serialize: JSON.stringify,
      parseValue: JSON.parse,
```

File: compass/src/store/index.js [hunk 4]
```diff
@@ -59,9 +58,36 @@ export function createApolloClient(tenant, token) {
...
+export async function fetchSchema(link, setSchema) {
+  if (!link) return;
+
+  const schema = await introspectSchema(link);
+  const executableSchema = makeRemoteExecutableSchema({
+    schema,
+    link,
+  });
+  setSchema(executableSchema);
+}
```
Comment (Wawrzyn321) [inline compass/src/store/index.js]: I think this function should only receive a `link` and just return the `executableSchema` (as per it's name). In `ClientProvider`, where the function is used,  schema can be then assigned by
```
fetchSchema(link).then(executableSchema => setSchema(executableSchema));
```
Or just:
```
fetchSchema(link).then(setSchema);
```

**Comment by tgorgol** [comment]
We decided to go with a different approach to this issue.

**Pull request closed without merging**